filament/src/details/View.h

```diff
@@ -416,7 +416,9 @@ class FView : public View {
     mutable bool mHasDirectionalLight = false;
     mutable bool mHasDynamicLighting = false;
     mutable bool mHasShadowing = false;
+    ShadowMapManager mShadowMapManager;
     mutable ShadowMap mDirectionalShadowMap;
+    mutable std::array<std::unique_ptr<ShadowMap>, CONFIG_MAX_SHADOW_CASTING_SPOTS> mSpotShadowMap;
```

Comment:
Out of curiosity, why not use a `std::vector<ShadowMap>`  -- I'm not a fan of `unique_ptr`. Do we remove entries in the middle of the array often?

Reply:
I choose `std::array` because the size is known at compile time. And no, we don't add or remove entries once `FView` is constructed.

`ShadowMap` has to be constructed with a reference to `FEngine`, so the container has to store pointers.